src/test/ui/issues/issue-66958.rs

```diff
@@ -0,0 +1,15 @@
+// edition:2018
```

Comment:
Could you move the test into `ui/async-await/issues/` and also append a short description to the file name?